src/client/opamState.ml

```diff
@@ -1989,7 +1989,7 @@ let string_of_env_update t shell updates =
       | "+="
       | ":=" -> (ident, Printf.sprintf "%s:$%s" string ident)
       | "=:"
-      | "=+" -> (ident, Printf.sprintf "$%s:%s" ident string)
+      | "=+" -> (ident, (match shell with `csh -> Printf.sprintf "${%s}:%s" ident string | _ -> Printf.sprintf "$%s:%s" ident string))
```

Comment:
This line is changed since "$MANPATH:..." raises an error in (t)csh, where ":" means a modifier for the variable reference "$MANPATH".  (This is a second try, as the previous change didn't type-check in OCaml 3.x.)